### PR TITLE
feat(reranking extensions): Implement Reranking extensions settings

### PR DESCRIPTION
**Pull request by VladislavFitz**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | no
| New feature?      | yes
| BC breaks?        | no    
| Need Doc update   | yes

## Describe your change

Add reranking extension structure to the index settings

### CODE REVIEW

File: algolia/internal/gen/templates/settings.go.tmpl [hunk 1]
```diff
@@ -16,6 +16,7 @@ type Settings struct {
     {{- range $_, $setting := . }}
     {{ title $setting.Name }} *opt.{{ title $setting.Name }}Option `json:"{{ $setting.Name }},omitempty"`
     {{- end }}
+    Extensions *Extensions `json:"Extensions,omitempty"`
```
Comment (netgusto) [inline algolia/internal/gen/templates/settings.go.tmpl:19]: Is it expected that the json exposes an upper-cased name?

File: algolia/search/settings.go [hunk 2]
```diff
@@ -73,6 +73,7 @@ type Settings struct {
 	Advanced                                *opt.AdvancedOption                                `json:"advanced,omitempty"`
 	AttributeCriteriaComputedByMinProximity *opt.AttributeCriteriaComputedByMinProximityOption `json:"attributeCriteriaComputedByMinProximity,omitempty"`
 	UserData                                *opt.UserDataOption                                `json:"userData,omitempty"`
+	Extensions                              *Extensions                                        `json:"Extensions,omitempty"`
```
Comment (netgusto) [inline algolia/search/settings.go:76]: Same here :) why upper-case on the JSON ?

**Review by netgusto — COMMENTED** [review]
Some comments.

**Review by netgusto — APPROVED** [review]
Seen the changes, 👍 thank you!

**Comment by Jerska** [comment]
🎉 6

**Pull request merged**